### DISCUSSION TITLE
bookmarks manager looks different

### DISCUSSION THREAD

**Discussion by muellerto** [leo-arch/clifm#314] [General]
When I use the `bm` command to open the bookmarks manager it looks different on Windows and Linux. On Linux I get the paths behind the configured bookmarks:
```
Bookmarks Manager [Linux]

1 [c] clifm
2 [u] /home/muellerto
3 [w] /home/muellerto/work

Enter 'e' to edit your bookmarks or 'q' to quit
Choose a bookmark (by ELN, shortcut, or name):
>
```
This is very helpful if you know there was a damned bookmark but you can't remember the name or the spelling.

On Windows I get indeed only the bookmark names, no paths at all. And my names are rather short:
```
Bookmarks Manager [Windows]

1 clifm
2 u
3 w
4 t
5 doc            [<- would be 'd' if 'd' would be allowed]
6 p

Enter 'e' to edit your bookmarks or 'q' to quit
Choose a bookmark (by ELN, shortcut, or name):
>
```
Same version 1.21.8 from today.

**Comment by leo-arch** [comment]
Hi @muellerto. Here's the thing:

1. Bookmarks have three components: a path, a name, and (optionally) a shortcut (enclosed in square brackets). For example, in your Linux screenshot the first entry has a shortcut (`[c]`), a name `clifm`, and a path, while the remaining two entries have paths and shortcuts (but no names).
2. Paths are displayed only provided there's no name, which is why entries 2 and 3 show the paths.
3. In your Windows screenshot there are no shortcuts, only names, which is why paths are not displayed.

This assumes that bookmark names are descriptive enough (so that paths need not to be displayed). Maybe this is a bad assumption, and paths should always be displayed.

**Reply by leo-arch** [reply]
Give it a try now: paths are always displayed and TAB completion is available for bookmark names.

**Comment by muellerto** [comment]
Ah, this explains the behavior.

Your change is good. I couldn't explain myself why the paths are hidden when there are names. I find it's really much better now, you get an overview of the bookmarks and where they go at a glance.

I would say these bookmarks are even more meaningful on Windows because you have the damned drive name there you don't have on Linux. So I use bookmarks much more on Windows, this exempts me from remembering drives. In some cases I have equal directory names on different drives. So which shortcut I used for what path? Sometimes (after holidays) it's a good idea to get an overview again.

Thanks a lot for this.

**Reply by leo-arch** [reply]
You're welcome @muellerto.
